src/adapter/src/coord/message_handler.rs

```diff
@@ -427,6 +427,31 @@ impl<S: Append + 'static> Coordinator<S> {
     #[tracing::instrument(level = "debug", skip_all)]
     async fn message_compute_instance_status(&mut self, event: ComputeInstanceEvent) {
         event!(Level::TRACE, event = format!("{:?}", event));
+
+        let (instance_name, replica_name) = self
+            .catalog
+            .compute_instances()
+            .find(|i| i.id == event.instance_id)
+            .map(|i| {
+                let mut replica = event.replica_id.to_string();
+                for (name, id) in &i.replica_id_by_name {
+                    if *id == event.replica_id {
+                        replica = name.clone();
+                        break;
+                    }
+                }
+                (i.name.clone(), replica)
+            })
+            .unwrap_or_else(|| (event.instance_id.to_string(), event.replica_id.to_string()));
+
+        if matches!(event.status, ComputeInstanceStatus::NotReady) {
+            self.broadcast_notice(AdapterNotice::ClusterReplicaStatusChanged {
+                cluster: instance_name,
+                replica: replica_name,
+                status: event.status,
+            });
```

Comment:
We're not doing anything to limit the status updates to sessions that have an active query on this cluster, or have their `cluster` parameter set to this cluster, right? That seems fine to start, just wanted to check that that was intended.

Reply:
It limits to cluster messages for sessions on that cluster: https://github.com/MaterializeInc/materialize/pull/15262/files#diff-bd136ceec5bfc71f2d8800397ab50fdea6a17b01c754f3495d2387c0c80fd974R334

Reply:
Oh, just missed it! My bad.